File: front/src/client/bll/apis/participate.js
import config from '../../config/config.js'
import requestService from './common/request.js'
// import commonFunc from './common/common.js'
const participatenApi = {
  async getDeviceType () {
    let url = config.APISERVER.host + '/participate/getDeviceType'
    let res = await requestService.get(url, '')
    return res
  },
  async getIp () {
    let url = config.APISERVER.host + '/participate/getIp'
    let res = await requestService.get(url, '')
    return res
  },
  async getAddress () {
    let url = config.APISERVER.host + '/participate/getAddress'
    let res = await requestService.get(url, '')
    return res
  },
  async addNewParticipate (params) {
    let url = config.APISERVER.host + '/participate/addNewParticipate'
    let res = await requestService.post(url, params)
    return res
  },
  async getAll (params) {
    let url = config.APISERVER.host + '/participate/getAll'
    let res = await requestService.get(url, params)
    return res
  },
  async updateEndTime (params) {
    let url = config.APISERVER.host + '/participate/updateEndTime'
    let res = await requestService.post(url, params)
    return res
  },
  async selectParticiByTime (params) {
    let url = config.APISERVER.host + '/participate/selectParticiByTime'
    let res = await requestService.get(url, params)
    return res
  }
}
export default participatenApi

File: houduan/src/main/java/com/code/questionnaireSystem/controller/Participate.java
package com.code.questionnaireSystem.controller;

import java.util.Date;

import javax.servlet.http.HttpServletRequest;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.code.questionnaireSystem.service.ParticipateService;
import com.code.questionnaireSystem.utils.Result;

/**
 * 参与者（被调查者）controller层
 * 
 * @author 皮艳萍
 *
 */
@RestController
@RequestMapping("/participate")
public class Participate {
	@Autowired
	private ParticipateService participateService;

	@GetMapping("/getDeviceType")
	public Result getDeviceType(HttpServletRequest request) {
		return participateService.getDeviceType(request);

	}

	@GetMapping("/getIp")
	public Result getIp(HttpServletRequest request) {
		return participateService.getIp(request);
	}

	@GetMapping("/getAddress")
	public Result getAddress(HttpServletRequest request) throws Exception {
		return participateService.getAddress(request);
	}

	@GetMapping("/getAll")
	public Result getAll(String surverId) {
		return participateService.getAll(surverId);
	}

	@PostMapping("/addNewParticipate")
	public Result addNewParticipate(String ip, String address, String device, String surverId) {
		return participateService.addNewParticipate(ip, address, device, surverId);
	}

	@PostMapping("/updateEndTime")
	public Result updateEndTime(String participateId) {
		return participateService.updateEndTime(participateId);
	}

	@GetMapping("/selectParticiByTime")
	public Result selectParticiByTime(Date startTime, Date endTime, String surverId) {
		return participateService.selectParticiByTime(startTime, endTime, surverId);
	}

}
